src/main/java/org/openstreetmap/atlas/checks/validation/intersections/SelfIntersectingPolylineCheck.java

```diff
@@ -144,4 +170,36 @@ else if (object instanceof Area)
     {
         return FALLBACK_INSTRUCTIONS;
     }
+
+    /**
+     * Returns true if adjacent {@link Edge} has identical lat,lng sequences
+     *
+     * @param polyline
+     *            the {@link PolyLine} being examined
+     * @return {@code true} if the any set of adjacent edges have identical geometries
+     */
+    private boolean hasDuplicateEdges(final PolyLine polyline)
+    {
+        final List<Segment> segments = polyline.segments();
+        final List<Segment> duplicates = new ArrayList<>();
+
+        // Loop through Polyline Segments
+        for (int i = 0; i < segments.size(); i++)
```

Comment:
Please avoid using single letter variables to keep code style consistent with the rest of the code base.